### ISSUE TITLE
it will be ban or how can I add proxy to it?

### ISSUE THREAD

**Issue by septfish** [avilash/TikTokAPI-Python#11]

**Comment by thanatos11** [comment]
it will be banned if you do to many calls

**Comment by HERCULESxp** [comment]
@thanatos11  i did many calls and now i am banned the requests are illegal, the questions is i will be banned forever?

**Comment by HERCULESxp** [comment]
@thanatos11 sorry for my english i'm using google translater

**Comment by thanatos11** [comment]
> @thanatos11 i did many calls and now i am banned the requests are illegal, the questions is i will be banned forever?

the request are not illegal, they detect robots and dennies that to save their server performance. Now, if you made a lot of request, just logg in with yu user used, and do the capcha they require, then logg off, and login, get the new cookie and replace it. It should work, at least for my application, where i wanted to get user info, it works.

**Comment by thanatos11** [comment]
also, you can change the IP, they ban the IP

**Comment by VivianoRiccardo** [comment]
Yes they ban the ip i make something like 300 requests per hour and they ban me, but after a while they remove the ban. The options are: use proxy rotation, use vpn, implement a sleep component for the requests, or as thantos suggested you need to set new cookies in the header file. At the moment i m trying vpn
